discovery/discovery.go

```diff
@@ -109,12 +110,15 @@ func (o *orchestratorPool) GetOrchestrators(ctx context.Context, numOrchestrator
 		}
 		return caps.CompatibleWith(info.Capabilities)
 	}
+	var wg sync.WaitGroup
 	getOrchInfo := func(ctx context.Context, od common.OrchestratorDescriptor, infoCh chan common.OrchestratorDescriptor, errCh chan error) {
+		defer wg.Done()
```

Comment:
```suggestion
ctx, cancel := context.WithTimeout(clog.Clone(context.Background(), ctx), maxGetOrchestratorCutoffTimeout)
defer cancel()
```

Related to the previous comment. This should eliminate the need for having this waitgroup.